### PR TITLE
tiltfile: added error when duplicate yaml is detected

### PR DESCRIPTION
**Pull request by abdullahzameek**
Hi @maiamcc @jazzdan, 

Please review the following commits I made in branch abdullahzameek/3588-duplicate-yaml:

0b8e13f234d1c34c1eeeb3129f2512978c523b77 (Tue Jul 21 17:27:42 2020 +0400)
tiltfile: added error when duplicate yaml is detected

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/k8s/names.go [hunk 1]
```diff
@@ -5,8 +5,14 @@ import (
 	"strings"
 )
 
+const FmtduplicateYamlDetectedError = "Duplicate YAML Entity(s): [%s] has been detected across one or more resources.  Only one specification per entity can be applied to the cluster; to ensure expected behavior, remove the duplicate specifications."
```
Comment (jazzdan) [inline internal/k8s/names.go:8]: Does this need to be exported? (It's exported because it begins with an uppercase letter)

Comment (abdullahzameek) [reply internal/k8s/names.go:8]: Yep! I'm using that `const` and function in `tiltfile_test.go` in `loadErrString()`

File: internal/tiltfile/tiltfile_test.go [hunk 2]
```diff
@@ -3836,57 +3837,61 @@ k8s_yaml('foo.yaml')
...
-	f.assertWarnings(duplicateWarningStr)
+	labelsFoo := map[string]string{"foo": "bar"}
+	f.yaml("all.yaml",
+		deployment("foo-deployment", image("gcr.io/foo1"), namespace("ns1"), withLabels(labelsFoo)),
+		service("foo-service", withLabels(labelsFoo)),
+		service("foo-service", withLabels(labelsFoo)))
+	f.file("Tiltfile", `
+k8s_yaml('all.yaml')
+`)
+	f.loadErrString(k8s.DuplicateYamlDetectedError("foo-service:service:default:core"))
```
Comment (abdullahzameek) [inline internal/tiltfile/tiltfile_test.go:3880]: @jazzdan This seemed to be a good way to get access to that const....should I re-declare it in this file, or is the current method okay?

File: internal/k8s/names.go (hunk 1 above)
Comment (maiamcc) [inline internal/k8s/names.go:8]: This is a very silly wording correction but the current implementation will only ever error about one duplicate at a time, so you don't need `entity(s)`, just `entity` is fine
```suggestion
const FmtduplicateYamlDetectedError = "Duplicate YAML Entity: "%s" has been detected across one or more resources.  Only one specification per entity can be applied to the cluster; to ensure expected behavior, remove the duplicate specifications."
```

File: internal/tiltfile/tiltfile_state.go [hunk 3]
```diff
@@ -654,7 +651,14 @@ func (s *tiltfileState) assembleK8sV1() error {
 }
 
 func (s *tiltfileState) assembleK8sV2() error {
-	err := s.assembleK8sByWorkload()
+	//Note: We're using UniqueNames() over here just to check for duplicate resources, hence
```
Comment (maiamcc) [inline internal/tiltfile/tiltfile_state.go:654]:
```suggestion
	//Note: We're using UniqueNames() over here just to check for duplicate YAML entities, hence
```

File: internal/tiltfile/tiltfile_test.go (hunk 2 above)
Comment (jazzdan) [reply internal/tiltfile/tiltfile_test.go:3880]: Ah! I missed this. Yeah, makes sense! My question was a genuine one, not a gotcha :sweat_smile:

File: internal/k8s/names.go (hunk 1 above)
Comment (maiamcc) [reply internal/k8s/names.go:8]: hmm, I don't see `FmtduplicateYamlDetectedError` used anywhere but this file; this const can be private but accessed by a public function `DuplicateYamlDetectedError` elsewhere!

Small naming sidenote that doesn't super matter: I think you want `fmtDuplicate...` (capital D), and in general acronyms/initialisms like `YAML` should be [consistently cased](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) -- so this func should be `fmtDuplicateYAMLDetectedError`

File: internal/tiltfile/tiltfile_state.go (hunk 3 above)
Comment (abdullahzameek) [reply internal/tiltfile/tiltfile_state.go:654]: Aaaaa I keep making this mistake of referring to YAML entities as resources haha

File: internal/k8s/names.go (hunk 1 above)
Comment (abdullahzameek) [reply internal/k8s/names.go]: Ohhh right, I thought I wouldn't be able to access a private const, even through a public function. Will correct that and the casing!

**Comment by abdullahzameek** [comment]
@landism, 

@maiamcc and I discussed that since this PR has code that changes the way duplicate YAML entities are handled entirely from the previous approach (appending a numerical index to the end of the entity name), this might be a breaking change in this week's release?

File: internal/k8s/names.go (hunk 1 above)
Comment (abdullahzameek) [reply internal/k8s/names.go]: Corrected this, will merge into master now!

**Comment by abdullahzameek** [comment]
Merged into master!

**Comment by jazzdan** [comment]
Wohoo! Thanks @abdullahzameek

**Pull request merged**